src/server/verification/processor/provider/ZoomProvider.js

```diff
@@ -31,83 +38,130 @@ class ZoomProvider implements IEnrollmentProvider {
+      await api.readEnrollment(enrollmentIdentifier, customLogger)
+      alreadyEnrolled = true
+    } catch (exception) {
+      // if something other that 'FacemapNotFound was thrown - re-throwing
+      if (FacemapNotFound !== exception.name) {
+        throw exception
+      }
 
-      faceSearchResponse = response
-      // excluding own enrollmentIdentifier
-      duplicate = results.find(
-        ({ enrollmentIdentifier: matchId }) => matchId.toLowerCase() !== enrollmentIdentifier.toLowerCase()
-      )
+      // otherwise, setting alreadyEnrolled to false
+      alreadyEnrolled = false
+    }
+
+    // 2. performing liveness check and storing facescan / audit trail images (if need)
+    let isLive = true
+
+    try {
+      const callArgs = [enrollmentIdentifier, payload, customLogger]
+      const enroll = () => api.submitEnrollment(...callArgs)
+
+      // if already enrolled, will call checkLiveness
+      // this is need because enroll doesn't re-checks facemap/images
+      // for liveness if identifier already enrolled
+      if (alreadyEnrolled) {
+        // it doesn't need enrollmentIdentifier param,
+        // so we're slicing it from the start of the args
+        await api.checkLiveness(...slice(callArgs, 1))
+
+        // if api supports delete enrollment endpoint
+        if (isDisposeSupported) {
+          // then we're refreshing enrolled facemap with the data just received
+          // firstly, removing snapshot enrolled before
+          await this.dispose(enrollmentIdentifier)
+          // then enrolling it with the new data already verified for liveness
+          await enroll()
+          // setting enrolled flag to false as we re-enrolled the facemap
+          alreadyEnrolled = false
+        }
+      } else {
+        // if not enrolled/stored yet - just calling enroll.
+        await enroll()
+      }
     } catch (exception) {
-      faceSearchResponse = exception.response || {}
+      const { name, message, response } = exception
 
-      // if liveness issues were detected
-      if ('livenessCheckFailed' === faceSearchResponse.subCode) {
-        const isLive = false
-        const { message } = faceSearchResponse
+      // if liveness / security issues were detected
+      if ([LivenessCheckFailed, SecurityCheckFailed].includes(name)) {
+        isLive = false
 
         // notifying about liveness check failed
         await notifyProcessor({ isLive })
         log.warn(message, { enrollmentIdentifier })
-
-        throwCustomException(message, { isLive }, faceSearchResponse)
+        throwCustomException(message, { isLive }, response)
       }
 
-      // otherwisw just re-throw exception and stop processing
+      // otherwisw just re-throwing exception and stopping processing
       throw exception
     }
 
+    // notifying about liveness passed or not
+    await notifyProcessor({ isLive })
+
+    // 3. checking for duplicates
+    const { results, ...faceSearchResponse } = await api.faceSearch(
```

Comment:
i dont understand, it seems like you rewrote the logic and order, please revert

Reply:
Yes i rewrote everything and this is mandatory because v9 API was **totally refactored** by the Facetec and the old flow we used in v8 isn't supported anymore. The new flow is described some comments above

Reply:
reimplemented according suggestions in #290